java/test/jmri/jmrit/logixng/actions/WebRequestTest.java

```diff
@@ -168,21 +167,48 @@ public void testThrowTurnouts() throws JmriException {
 
         _responseCodeVariable.setValue(null);
         _replyVariable.setValue(null);
-        InstanceManager.getDefault(TurnoutManager.class).getByUserName("MiamiWest").setState(Turnout.THROWN);
+        InstanceManager.getDefault(TurnoutManager.class).provide("MiamiWest").setState(Turnout.THROWN);
```

Comment:
Why this change?

Reply:
These lines were being flagged as potential NPEs by tests-spotbugs as getByUserName could potentially return null.
provide does not return null, it throws an exception instead.

An alternative to this would add an assertion after each getByUserName checking that the value is not null.
I could do it this way if better for the test?

Reply:
> An alternative to this would add an assertion after each getByUserName checking that the value is not null.
> I could do it this way if better for the test?

Yes, thanks. The problem with `provide()` is that it tries to create the bean if the bean cannot be found. If the turnout `MiamiWest` is missing, it tries to create the turnout `ITMiamiWest`. This test listens on the turnouts and when a turnut is thrown, it does a web request with the name of the turnout. If the turnout has the wrong name, the web request will fail.

So for this test class, it's better to keep `getByUserName()` for all the tests and add an assertion after each call.